lib/hawkbit/hawkbit.c

```diff
@@ -0,0 +1,1237 @@
+
+	LOG_DBG("id=%s", log_strdup(d->id));
+	LOG_DBG("download=%s", log_strdup(d->deployment.download));
+	LOG_DBG("update=%s", log_strdup(d->deployment.update));
+	LOG_DBG("chunks[0].part=%s", log_strdup(c->part));
+	LOG_DBG("chunks[0].name=%s", log_strdup(c->name));
+	LOG_DBG("chunks[0].version=%s", log_strdup(c->version));
+	LOG_DBG("chunks[0].artifacts[0].filename=%s",
+		log_strdup(a->filename));
+	LOG_DBG("chunks[0].artifacts[0].hashes.sha1=%s",
+		log_strdup(a->hashes.sha1));
+	LOG_DBG("chunks[0].artifacts[0].hashes.md5=%s",
+		log_strdup(a->hashes.md5));
+	LOG_DBG("chunks[0].artifacts[0].hashes.sha256=%s",
+		log_strdup(a->hashes.sha256));
+	LOG_DBG("chunks[0].size=%d", a->size);
+	LOG_DBG("download-http=%s",
+		log_strdup(l->download_http.href));
+	LOG_DBG("md5sum =%s", log_strdup(l->md5sum_http.href));
+}
+
+int hawkbit_init(void)
+{
+	bool image_ok;
+	int ret = 0, rc = 0;
+	struct flash_pages_info info;
+	struct hawkbit_device_acid init_action_id;
+
+	fs.offset = DT_FLASH_AREA_STORAGE_OFFSET;
+	rc = flash_get_page_info_by_offs(device_get_binding(DT_FLASH_DEV_NAME),
+					 fs.offset, &info);
+	if (rc) {
+		LOG_ERR("Unable to get storage page info");
+		return -EIO;
+	}
+
+	fs.sector_size = info.size;
+	fs.sector_count = 3U;
+
+	rc = nvs_init(&fs, DT_FLASH_DEV_NAME);
+	if (rc) {
+		LOG_ERR("Storage flash Init failed");
+		return -ENODEV;
+	}
+
+	rc = nvs_read(&fs, ADDRESS_ID, &init_action_id, sizeof(init_action_id));
+	LOG_DBG("ACID: current %d, update %d", init_action_id.current,
+		init_action_id.update);
+
+	image_ok = boot_is_img_confirmed();
+	LOG_INF("Image is %s confirmed OK", image_ok ? "" : " not");
+	if (!image_ok) {
+		ret = boot_write_img_confirmed();
+		if (ret < 0) {
+			LOG_ERR("Couldn't confirm this image: %d", ret);
+			return ret;
+		}
+
+		LOG_DBG("Marked image as OK");
+		ret = boot_erase_img_bank(DT_FLASH_AREA_IMAGE_1_ID);
+		if (ret) {
+			LOG_ERR("Failed to erase second slot");
+			return ret;
+		}
+
+		if (init_action_id.update != -1) {
+			ret = hawkbit_device_acid_update(
+				HAWKBIT_ACTION_ID_CURRENT,
+				init_action_id.update);
+		}
+	}
+
+	return ret;
+}
+
+static void response_cb(struct http_response *rsp,
+			enum http_final_call final_data,
+			enum hawkbit_http_request type)
+{
+	int ret;
+	int downloaded;
+	u8_t *body_data = NULL;
+	static size_t body_len;
+	static size_t response_buffer_size = RESPONSE_BUFFER_SIZE;
+
+	switch (type) {
+	case HAWKBIT_PROBE:
+		if (hb_context.dl.http_content_size == 0) {
+			body_data = rsp->body_start;
+			body_len = strlen(rsp->recv_buf);
+			body_len -= (rsp->body_start - rsp->recv_buf);
+			strncpy(hb_context.response_data, body_data, body_len);
+			hb_context.dl.http_content_size = rsp->content_length;
+		}
+
+		if (body_data == NULL) {
+			if (final_data == HTTP_DATA_MORE) {
+				body_data = rsp->recv_buf;
+				body_len = body_len + (rsp->data_len);
+				if (body_len > response_buffer_size) {
+					response_buffer_size +=
```

Comment:
Can you elaborate the reason for adding `response_buffer_size` to itself here ? If this is not a logical bug, i would suggest this alternative, 
```
response_buffer_size <<= 1;
```
This would get rid of the useless line break and is also slightly more efficient that addition..